### PR TITLE
Credit Card Endpoints

### PR DESCRIPTION
**Pull request by ZweWaiYanHtet**

### CODE REVIEW

File: pokepaylib/src/main/java/jp/pokepay/pokepaylib/PartnerAPI/PartnerRequest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package jp.pokepay.pokepaylib.PartnerAPI;
+
+import java.util.Map;
+
+import jp.pokepay.pokepaylib.BankAPI.BankRequestError;
+import jp.pokepay.pokepaylib.OAuthAPI.OAuthRequestError;
+import jp.pokepay.pokepaylib.ProcessingError;
+import jp.pokepay.pokepaylib.Request;
+
+public abstract class PartnerRequest {
```
Comment (cxxxr) [inline pokepaylib/src/main/java/jp/pokepay/pokepaylib/PartnerAPI/PartnerRequest.java:10]: When you say "Partner" in pokepay, it is confusing to think of the Partner API.
What about other expressions? (e.g. external)?

File: pokepaylib/src/main/java/jp/pokepay/pokepaylib/BankAPI/autogen/requests/GetCreditCards.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    private String after;
+    private Integer perPage;
+    private String organizationCode;
+
+    public GetCreditCards(String userId, String organizationCode) {
+        this.userId = userId;
+        this.organizationCode = organizationCode;
+    }
+
+    public GetCreditCards before(String before) {
```
Comment (ZweWaiYanHtet) [inline pokepaylib/src/main/java/jp/pokepay/pokepaylib/BankAPI/autogen/requests/GetCreditCards.java:25]: Why is the naming for these optional setter functions different if they are autogenerated?  It was `setCardholderNameSpecified` in CreateCreditCard.  Shouldn't this be `setBefore`?

File: pokepaylib/src/main/java/jp/pokepay/pokepaylib/JsonConverter.java [hunk 3]
```diff
@@ -22,10 +22,9 @@ public static ObjectMapper createObjectMapper() {
                 .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
     }
 
-    public static String toString(final Map<String, Object> map)
-            throws JsonGenerationException, JsonMappingException, IOException {
-        final JSONObject j = new JSONObject(map);
-        return j.toString();
+    public static String toString(final Map<String, Object> map) throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        return objectMapper.writeValueAsString(map);
```
Comment (ZweWaiYanHtet) [inline pokepaylib/src/main/java/jp/pokepay/pokepaylib/JsonConverter.java:27]: May I know why this got changed?

Comment (cxxxr) [reply pokepaylib/src/main/java/jp/pokepay/pokepaylib/JsonConverter.java:27]: The purpose is to be able to run tests with JUnit without android simulator.

File: pokepaylib/src/main/java/jp/pokepay/pokepaylib/PartnerAPI/ExternalServiceRequest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package jp.pokepay.pokepaylib.PartnerAPI;
```
[the remark this answers is not in the document]
Comment (cxxxr) [reply pokepaylib/src/main/java/jp/pokepay/pokepaylib/PartnerAPI/ExternalServiceRequest.java]: Oh, I overlooked it.
Thanks.

File: pokepaylib/src/main/java/jp/pokepay/pokepaylib/BankAPI/autogen/requests/GetCreditCards.java (hunk 2 above)
Comment (cxxxr) [reply pokepaylib/src/main/java/jp/pokepay/pokepaylib/BankAPI/autogen/requests/GetCreditCards.java:25]: Generally, I think that a builder pattern (which may not strictly be a builder pattern) is not a set, but only a name.

Comment (cxxxr) [reply pokepaylib/src/main/java/jp/pokepay/pokepaylib/BankAPI/autogen/requests/GetCreditCards.java:25]: For example
```
new GetCreditCards(userId, organizationCode)
  .after("xxx")
  .perPage(20);
```

Comment (ZweWaiYanHtet) [reply pokepaylib/src/main/java/jp/pokepay/pokepaylib/BankAPI/autogen/requests/GetCreditCards.java:25]: Then, shouldn't we change `setCardholderNameSpecified` to just `cardholderNameSpecified` or `isCardholderNameSpecified`?

Comment (cxxxr) [reply pokepaylib/src/main/java/jp/pokepay/pokepaylib/BankAPI/autogen/requests/GetCreditCards.java:25]: Oh, sorry.
This was not a response to your comment.
It seems to be a bug in the generator that malfunctions when the prefix of the openapi parameter name is "is".

Comment (cxxxr) [reply pokepaylib/src/main/java/jp/pokepay/pokepaylib/BankAPI/autogen/requests/GetCreditCards.java:25]: fixed it

**Review by DanRidh — APPROVED** [review]

**Pull request merged**